### PR TITLE
Implemented message signing by registered authorities in Circuit

### PR DESCRIPTION
**Pull request by zannis**

### CODE REVIEW

File: circuit/src/execution-delivery/src/message_assembly/substrate_gateway_assembly.rs [hunk 1]
```diff
@@ -99,144 +97,216 @@ where
...
+                version: 1,
+                signed_extensions: vec![DecodeDifferent::Decoded(String::from("test"))],
+            },
+            modules: DecodeDifferent::Decoded(vec![module_metadata]),
+        };
+        Metadata::new(runtime_metadata)
+    }
+
+    #[test]
+    #[should_panic(expected = "Module with a given name doesn't exist as per the current metadata")]
```
Comment (MaciejBaj) [inline circuit/src/execution-delivery/src/message_assembly/substrate_gateway_assembly.rs:204]: Runtime logic should never panic!, can u convert it to error throw?

**Review by MaciejBaj — CHANGES_REQUESTED** [review]

**Review by beqaabu — APPROVED** [review]
Looks good!

File: circuit/src/execution-delivery/src/message_assembly/chain_generic_metadata.rs [hunk 2]
```diff
@@ -48,29 +49,31 @@ impl Metadata {
...
-            .clone()
-            .find(|call| {
-                call_counter += 1;
-                call.name.clone() == DecodeDifferent::Encode(lookup_call_name)
-            })
-            .ok_or(MetadataError::CallNotFound(
-                "Call with a given name doesn't exist on that module as per the current metadata",
-            ))
-            .unwrap();
+        let _call_found = calls.clone().find(|call| {
```
Comment (MaciejBaj) [inline circuit/src/execution-delivery/src/message_assembly/chain_generic_metadata.rs:68]: remove "_" from var name when used

**Review by MaciejBaj — CHANGES_REQUESTED** [review]

File: circuit/src/execution-delivery/src/message_assembly/chain_generic_metadata.rs (hunk 2 above)
Comment (zannis) [reply circuit/src/execution-delivery/src/message_assembly/chain_generic_metadata.rs]: Fixed on following commit.

File: circuit/src/execution-delivery/src/message_assembly/substrate_gateway_assembly.rs [hunk 3]
```diff
@@ -320,7 +320,10 @@ pub mod tests {
 
             let expected_message: Vec<u8> = vec![
                 160, 1, 2, 12, 0, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-                1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 12, 8,
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 12, 8, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 9, 9,
```
Comment (MaciejBaj) [inline circuit/src/execution-delivery/src/message_assembly/substrate_gateway_assembly.rs:323]: 👌

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**